tracks-sw.js

```diff
@@ -19,47 +18,43 @@ d3.json(filename01)
                         .domain([xMin,xMax])
                         .range([0,1000]);
       var xScale = 1000/xMax;
-      var xStart = xScale*xArray[0];
-      var svg = d3.select('svg');
+      var xStart = xScale*beatListArray[0];
       var xAxis = d3.axisBottom().scale(axisScale);
 
-      svg.selectAll('g')
-        .data(xArray)
-        .enter().append('g')
-          .attr('transform', function(d) {return 'translate(0,30)';})
+      var svg = d3.select('svg');
+      var trackLineGroup = svg.append('g')
```

Comment:
You were binding the beat position list to an SVG group ("g") and then within that group, you would bind it to a SVG lines ("line").  This created one group for each beat and each group would have as many lines as beats.  So if you had 100 beats, you would have 100x100=10,000 SVG elements -- which made the program very slow.  We can talk about SVG elements and functions later.